tmt/steps/execute/__init__.py

```diff
@@ -70,17 +75,18 @@ def load(self, extra_keys=None):
         except tmt.utils.FileError:
             self.debug('Test results not found.', level=2)
 
-    def save(self, data=None):
+    def save(self, data: Optional[tmt.steps.StepData] = None) -> None:
         """ Save test results to the workdir """
         data = data or {}
         super().save(data)
         results = dict([
             (result.name, result.export()) for result in self.results()])
         self.write('results.yaml', tmt.utils.dict_to_yaml(results))
 
-    def _map_old_methods(self):
+    def _map_old_methods(self) -> None:
         """ Map the old execute methods in a backward-compatible way """
         how = self.data[0]['how']
+        assert isinstance(how, str)
```

Comment:
I wonder, what is the problem here. I would expect `self.data` to be `tmt.steps.StepData`, which is

```
   71 class StepData(TypedDict, total=False):
   72     """
   73     Step data structure
   74     """
   75     name: str
   76     how: str
   77     order: Optional[int]
```

So why we need to assert it here?

Reply:
assert was removed, thanks, resolving